Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -180,19 +182,61 @@ protected override void OnCreate(Bundle savedInstanceState)
 			OnStateChanged();
 
 			AddStatusBarUnderlay();
+			Window.DecorView.ViewTreeObserver.GlobalLayout += ViewTreeObserverOnGlobalLayout;
 		}
 
 		protected override void OnDestroy()
 		{
 			// may never be called
 			base.OnDestroy();
 
+			Window.DecorView.ViewTreeObserver.GlobalLayout -= ViewTreeObserverOnGlobalLayout;
+
 			MessagingCenter.Unsubscribe<Page, AlertArguments>(this, Page.AlertSignalName);
 			MessagingCenter.Unsubscribe<Page, bool>(this, Page.BusySetSignalName);
 			MessagingCenter.Unsubscribe<Page, ActionSheetArguments>(this, Page.ActionSheetSignalName);
 
-			if (_platform != null)
-				_platform.Dispose();
+			_platform?.Dispose();
+		}
+
+		void ViewTreeObserverOnGlobalLayout(object sender, EventArgs eventArgs)
+		{
+			var rect = new Rect();
+			Window.DecorView.GetWindowVisibleDisplayFrame(rect);
+
+			int height = Window.DecorView.Context.Resources.DisplayMetrics.HeightPixels;
+			int diff = height - rect.Bottom;
+
+			if (diff != 0)
+			{
+				// set status bar color so status controls have a background
+				if (Forms.IsLollipopOrNewer)
+					Window.SetStatusBarColor(((ColorDrawable)_statusBarUnderlay.Background).Color);
```

Comment:
Currently, when the soft keyboard shows, the user sees the system tray controls without a background. This is quite annoying. Here, I set status bar background so that the controls have a back color.